src/js/App/GlobalFilter/GlobalFilter.js

```diff
@@ -1,211 +1,68 @@
-            setValue(() =>
-              [...(selected || []), ...(sidSelected || [])].reduce(
-                (acc, { namespace, key, value }) =>
-                  updateSelected(acc, namespace, `${key}${value ? `=${value}` : ''}`, value, true, { item: { tagKey: key } }),
-                selectedTags
-              )
-            );
-          }}
-        />
-      )}
-      <Divider />
-    </Fragment>
+  return useCallback(
+    (activeTags, search) => {
+      storeFilter(activeTags, hasAccess && !isDisabled, history);
+      batch(() => {
+        dispatch(
+          fetchAllTags({
+            registeredWith,
+            activeTags,
+            search,
+          })
+        );
+        dispatch(
+          fetchAllSIDs({
+            registeredWith,
+            activeTags,
+            search,
+          })
+        );
+        dispatch(
+          fetchAllWorkloads({
+            registeredWith,
+            activeTags,
+            search,
+          })
+        );
+      });
+    },
+    [registeredWith, hasAccess, history]
   );
 };
 
-GlobalFilterDropdown.propTypes = {
-  isAllowed: PropTypes.func.isRequired,
-  isDisabled: PropTypes.bool,
-  userLoaded: PropTypes.bool,
-  filter: PropTypes.object,
-  chips: PropTypes.arrayOf(
-    PropTypes.shape({
-      category: PropTypes.string.isRequired,
-      chips: PropTypes.arrayOf(
-        PropTypes.shape({ key: PropTypes.string.isRequired, tagKey: PropTypes.string.isRequired, value: PropTypes.node.isRequired })
-      ),
-    })
-  ),
-  setValue: PropTypes.func.isRequired,
-  selectedTags: PropTypes.object,
-  isOpen: PropTypes.bool,
-  filterTagsBy: PropTypes.string,
-  filterScope: PropTypes.string,
-  setIsOpen: PropTypes.func.isRequired,
-};
-
-const GlobalFilter = () => {
-  const [hasAccess, setHasAccess] = useState(undefined);
-  const firstLoad = useRef(true);
-
-  const isAllowed = () => hasAccess;
-  const history = useHistory();
-
+const GlobalFilter = ({ hasAccess }) => {
   const [isOpen, setIsOpen] = useState(false);
-  const [token, setToken] = useState();
   const dispatch = useDispatch();
-  const { isLoaded, count, total, sapCount, aapCount, mssqlCount, isDisabled } = useSelector(
-    ({ globalFilter: { tags, sid, workloads, globalFilterHidden }, chrome: { appId } }) => ({
-      isLoaded: tags.isLoaded && sid.isLoaded && workloads.isLoaded,
+  const isLoaded = useSelector(({ globalFilter: { tags, sid, workloads } }) => tags.isLoaded && sid.isLoaded && workloads.isLoaded);
+  const { count, total, tags, sid, workloads } = useSelector(
+    ({ globalFilter: { tags, sid, workloads } }) => ({
       count: tags.count + sid.count + workloads.count,
       total: tags.total + sid.total + workloads.total,
-      sapCount: workloads.hasSap,
-      aapCount: workloads.hasAap,
-      mssqlCount: workloads.hasMssql,
-      isDisabled: globalFilterHidden || !appId,
+      tags: tags.items || [],
+      sid: sid.items || [],
+      workloads,
```

Comment:
Why do we not return the workload items? Does use a different data structure?

Reply:
Yup, workloads have slightly different data structure.